proposals/0000-specify-dump-json/schema.json

```diff
@@ -0,0 +1,115 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Dump JSON",
+  "description": "JSON dump of the GHC compiler diagnostics output",
+  "type": "object",
+  "properties": {
+    "version": {
+      "description": "The current JSON schema version of the JSON dump",
+      "type": "string"
+    },
+    "diagnostics": {
+      "description": "The list of diagnostics produced by GHC",
+      "type": "array",
+      "items": {
+        "properties": {
+          "span": {
+            "$ref": "#/$defs/span"
+          },
+          "severity": {
+            "description": "The diagnostic severity",
+            "type": "string",
+            "enum": [
+              "Warning",
+              "Error"
+            ]
+          },
+          "code": {
+            "description": "The diagnostic code (if it exists)",
+            "type": [
+              "integer",
+              "null"
+            ]
+          },
+          "hints": {
+            "description": "The hints suggested by GHC",
+            "type": "array"
+          },
+          "message": {
```

Comment:
> One thing I'm surprised about is that the diagnostic itself is represented as a string, rather than anything more structured. While including the string is sensible, I can imagine tools wanting to analyse the content. Of course there is a challenge here in that the types representing possible diagnostics have many constructors and are subject to change, so it isn't obvious how much they should be baked into the schema. But we could at least have a field containing a JSON-serialised version of the internal GHC diagnostic, and perhaps we could even generically derive the schema itself (so each GHC version could ship with its particular schema).

To make sure I am understanding you correctly, you mean that this particular field could be a JSON serialized version of the diagnostic message, rather than the message text itself? That is a great point. I believe that that should be added in addition to the text version. 

Any stringent requirements dictated on the schema on the shape of the data in such a field does seem really difficult to maintain, especially considering that the "errors-as-values" proposal is yet to be fully implemented. Agreed, that having a field that contains the JSON-serialized error with no schema constraint is a good idea. Thanks for the suggestions!